tests/platform_helper/providers/test_config.py

```diff
@@ -220,11 +78,13 @@ def test_validate_platform_config_succeeds_if_pipeline_account_matches_environme
         }
     }
 
-    # Should not error if config is sound.
-    config_provider = ConfigProvider(ConfigValidator())
+    mock_file_provider = Mock(spec=FileProvider)
+    mock_file_provider.load.return_value = platform_env_config
+    config_provider = ConfigProvider(ConfigValidator(), mock_file_provider)
     config_provider.config = platform_env_config
 
-    config_provider.validate_platform_config()
+    # Should not error if config is sound.
+    config_provider._validate_platform_config()
```

Comment:
```suggestion
    config_provider.load_and_validate_platform_config()
```
since we're using the file provider mock, we need to include the load part in this test, otherwise the mocked config will not be loaded into the config attribute yet

Reply:
We have considered loading config up front in the __init__ of the config provider which might provide a more intuitive structure to the class, but it was a disruptive change as we're suddenly getting errors the moment the ConfigProvider is initialised everywhere so hundreds of tests break, hence not done it yet.  But please comment  if you think that would be a good idea and we could pick it up again.

Reply:
Ah - thought I'd done that, but must have missed it...